rules/utils/should-add-parentheses-to-logical-expression-child.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+/**
+Check if parentheses should be added to a `node` when it's used as child of `LogicalExpression`.
+@param {Node} node - The AST node to check.
+@param {{operator: string, property: string}} options - Options
+@returns {boolean}
+*/
+function shouldAddParenthesesToLogicalExpressionChild(node, {operator, property}) {
+	// Not really needed, but more readable
+	if (
+		node.type === 'AwaitExpression' ||
+		node.type === 'BinaryExpression'
+	) {
+		return true;
+	}
+
+	// Lower precedence than `LogicalExpression`
+	// see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table
+	if (
+		node.type === 'ConditionalExpression' ||
+		node.type === 'AssignmentExpression' ||
+		node.type === 'AssignmentExpression' ||
+		node.type === 'YieldExpression' ||
+		node.type === 'SequenceExpression'
+	) {
+		return true;
+	}
+
+	/* istanbul ignore next: When operator or property is different, need more logic here, not implemented */
+	if (operator !== '??' || property !== 'left') {
+		throw new Error('Not supported.');
```

Comment:
Shouldn't we throw this exception at the beginning of the function so people don't try to use it for this without implementing it fully?

Reply:
I was hoping this function can return earlier, and if the operator or property is different. logic should be write here, but your concern make sense too, I don't have strong opinion.

Reply:
I think it's generally better to throw early. Otherwise, someone might use this, and not test it fully, and a consumer may hit the exception in the wild.